gtsfm/runner/frontend_runner.py

```diff
@@ -0,0 +1,70 @@
+"""Runner for the front-end only (when back-end results are not desired or required).
+
+Author: John Lambert
+"""
+from typing import Dict, Tuple
+
+import dask
+from dask.delayed import Delayed
+
+from gtsfm.feature_extractor import FeatureExtractor
+from gtsfm.loader.loader_base import LoaderBase
+from gtsfm.two_view_estimator import TwoViewEstimator
+
+
+def run_frontend(
```

Comment:
My most important comment here is why is this a function? My understanding is that a `runner` refers to a class with a run() method. Can we stick to that design ?

Reply:
I'm a bit ambivalent about this. We could do:
```python
class SceneFrontendOptimizer:
     def run(
        self, loader: LoaderBase, feature_extractor: FeatureExtractor, two_view_estimator: TwoViewEstimator
     ) -> Tuple[List[Keypoints], Dict[Tuple[int,int], Rot3], Dict[Tuple[int,int], Unit3], Dict[Tuple[int,int], np.ndarray]]:
          """ """"
```
or
```python
@dataclass
class SceneFrontendOptimizer:
     loader: LoaderBase
     feature_extractor: FeatureExtractor
     two_view_estimator: TwoViewEstimator

     def run(self) -> Tuple[List[Keypoints], Dict[Tuple[int,int], Rot3], Dict[Tuple[int,int], Unit3], Dict[Tuple[int,int], np.ndarray]]:
          """ """"
```
or we can keep it purely functional, as it is now (1 line of code instead of 2 lines, since we would have to instantiate the object).

Reply:
I think design consistency is more important. It helps to not have exceptions and let all runners be classes with some common structure. But it looks like this is already not the case (the argoverse runner is not a class). 
Other runners always construct a scene optimizer (there is a virtual method for it), but here we are not doing so. If we can change that about runners, we could have a base class design that could be used for this frontend runner as well. Otherwise it does not make much difference. We could make it a class so that users do the same thing for all runners, but since it is not derived from the runner base class anyway, I don't feel strongly about it.

Reply:
Thanks Akshay. Good points. I agree more consistency is better. I think we didn't make the Argoverse runner a child of the base runner since the arguments were so different. @ayushbaid is that right?

I'll merge for now, but let's keep in mind how we can improve this in the long run.

Reply:
I think having this as a class would have been consistent with our design policies thus far.